### PR TITLE
feat: add ListState / TableState scroll_down_by() / scroll_up_by() methods

### LINKED ISSUE #1207: Add scroll by methods to list and table
## Problem
Implement PageUp / Down like functionality
Suggested via https://github.com/intuis/rustmission/issues/13

## Solution
- work out exactly how to name this (not sure if this should be scroll_by(i32) or scroll_up_by(u16)/scroll_down_by(u16)
- implement

### PR DESCRIPTION
**Pull request by josueBarretogit**
Implement new methods `scroll_down_by(u16)` and `scroll_up_by(u16)` for both `Liststate` and `Tablestate`
Closes: #1207

### CODE REVIEW

**Review by orhun — APPROVED** [review]
General approval, I like this functionality 👍🏼

File: src/widgets/list/state.rs [hunk 1]
```diff
@@ -216,6 +216,54 @@ impl ListState {
...
+    /// ```
+    pub fn scroll_down_by(&mut self, amount: u16) {
+        match self.selected {
+            Some(index) => {
+                self.select(Some(index.saturating_add(amount as usize)));
+            }
+            None => {
+                self.select(Some(amount as usize));
+            }
+        }
```
Comment (joshka) [inline src/widgets/list/state.rs:241]:
```suggestion
        let selected = self.selected.unwrap_or_default();
        self.select(Some(selected.saturating_add(amount as usize)));
```

File: src/widgets/list/state.rs [hunk 2]
```diff
@@ -216,6 +216,54 @@ impl ListState {
...
+    /// ```
+    pub fn scroll_up_by(&mut self, amount: u16) {
+        match self.selected {
+            Some(index) => {
+                self.select(Some(index.saturating_sub(amount as usize)));
+            }
+            None => {
+                self.select(Some(0));
+            }
+        }
```
Comment (joshka) [inline src/widgets/list/state.rs:265]:
```suggestion
        let selected = self.selected.unwrap_or_default();
        self.select(Some(selected.saturating_sub(amount as usize)));
```

File: src/widgets/table/table_state.rs [hunk 3]
```diff
@@ -241,6 +241,54 @@ impl TableState {
...
+    /// ```
+    pub fn scroll_down_by(&mut self, amount: u16) {
+        match self.selected {
+            Some(index) => {
+                self.select(Some(index.saturating_add(amount as usize)));
+            }
+            None => {
+                self.select(Some(amount as usize));
+            }
+        }
```
Comment (joshka) [inline src/widgets/table/table_state.rs:266]:
```suggestion
        let selected = self.selected.unwrap_or_default();
        self.select(Some(selected.saturating_add(amount as usize)));
```

File: src/widgets/table/table_state.rs [hunk 4]
```diff
@@ -241,6 +241,54 @@ impl TableState {
...
+    /// ```
+    pub fn scroll_up_by(&mut self, amount: u16) {
+        match self.selected {
+            Some(index) => {
+                self.select(Some(index.saturating_sub(amount as usize)));
+            }
+            None => {
+                self.select(Some(0));
+            }
+        }
```
Comment (joshka) [inline src/widgets/table/table_state.rs:290]:
```suggestion
        let selected = self.selected.unwrap_or_default();
        self.select(Some(selected.saturating_sub(amount as usize)));
```

**Review by joshka — APPROVED** [review]
LGTM - with some simplifications to apply

**Comment by joshka** [comment]
merged to pick up fix for doc tests

**Comment by joshka** [comment]
Thanks for the PR :)

**Pull request merged**